### PR TITLE
Refactor Slice reference op for port to TFLM

### PR DESCRIPTION
**Pull request by annietllnd**
Solves second part of issue #51669, PR2:  Extract the reference implementation out of tensorflow/lite/kernels/internal/reference/reference_ops.h into its own header which can be included without dragging in reference_ops.h's dependences

Change-Id: I1c8e2a4550d9c6fd792b2d5ad0361579b9d437dd

### CODE REVIEW

**Review by advaitjain — CHANGES_REQUESTED** [review]
minor change requested then we're ready to merge. Please feel free to directly tag me on future PRs.

**Comment by annietllnd** [comment]
@advaitjain Thank you for reviewing, I'm currently working on fixing the Ubuntu Sanity build fail. Did you suggest any other changes for this patch? I am unable to find it in the changes view, therefore I wanted to double-check that I'm not missing anything. BR Annie

File: tensorflow/lite/kernels/internal/reference/slice.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_SLICE_H_
+#define TENSORFLOW_LITE_KERNELS_INTERNAL_REFERENCE_SLICE_H_
+
```
Comment (advaitjain) [inline tensorflow/lite/kernels/internal/reference/slice.h:17]: It appears that I didn't actually comment last time I reviewed.

Can you please add `#include` for the headers that declare SliceParams, RuntimeShape and SequentialTensorWriter?

That will ensure that slice.h follows https://google.github.io/styleguide/cppguide.html#Include_What_You_Use

**Comment by advaitjain** [comment]
> @advaitjain Thank you for reviewing, I'm currently working on fixing the Ubuntu Sanity build fail. Did you suggest any other changes for this patch? I am unable to find it in the changes view, therefore I wanted to double-check that I'm not missing anything. BR Annie

Apologies, looks like my previous comments got lost. Added a comment this time to IWYU.

The Ubuntu sanity build can sometimes fail for unrelated reasons. Once I approve the PR, it will be imported internally and that is a place where more exhaustive tests run.

**Review by advaitjain — APPROVED** [review]

**Comment by annietllnd** [comment]
> > @advaitjain Thank you for reviewing, I'm currently working on fixing the Ubuntu Sanity build fail. Did you suggest any other changes for this patch? I am unable to find it in the changes view, therefore I wanted to double-check that I'm not missing anything. BR Annie
> 
> Apologies, looks like my previous comments got lost. Added a comment this time to IWYU.
> 
> The Ubuntu sanity build can sometimes fail for unrelated reasons. Once I approve the PR, it will be imported internally and that is a place where more exhaustive tests run.

Thank you for the helpful reply and for your comments!

**Review by advaitjain — APPROVED** [review]

**Comment by advaitjain** [comment]
apologies for the delay. [`5a6b185` (#51672)](https://github.com/tensorflow/tensorflow/pull/51672/commits/5a6b185bab38c888a9d487cee9c773d2e7b759f9) fixes a lint error about not having a newline at the end of the file. This PR should be ready to go now.

**Pull request merged**